packages/react-form-state/src/FormState.ts

```diff
@@ -166,12 +166,18 @@ export default class FormState<
 
     this.setState({submitting: true});
 
-    const result = await onSubmit(formData);
-    if (result) {
-      this.updateRemoteErrors(result);
+    const errors = await onSubmit(formData);
+
+    if (errors) {
+      this.setState({submitting: false});
+      this.updateRemoteErrors(errors);
+    } else {
+      this.makeClean();
```

Comment:
how can this fix the issue mentioned in the PR if we're still waiting for the submit before getting here?

Reply:
submit wasn't stalling or anything the cache just wasn't updating properly